### PR TITLE
refactor: use a common filter for filters

### LINKED ISSUE #835: refactor: use a common filter component
### Refactor Request
I am opening this issue to have a record of it, I have already completed the task and will be opening the PR next

-- fill in the line below:
** discord username: el#0001 **

### PR DESCRIPTION
**Pull request by el**
# Description

Refactor for the commonly used filter components. Tthe time filter can be done next.

**discord username: afetharita#0001**

**closes #835**

Please describe your changes. Also describe your aim and content. Do not forget to list the dependencies required caused by those changes.

## Things to check before creating a PR
- [ ] I have inspected my topic, checked.
- [ ] If it is a core feature, executed detailed tests.

## Creating PR rules
- [ ] PR must be created for an issue with approved tag. Otherwise PR will be rejected.
- [ ] Relevant issue number: The issue number related to PR must be attached to head of PR header, after prefix after prefix # must be attached in parenthesis. A header like this could be used "prefix(#issue_number): PR header" .
- [ ] A descriptive and understandable title: The PR title should clearly describe the nature and purpose of the changes. The PR title should be the first thing displayed when the PR is opened. And it should follow the semantic commit rules. For example, a title like "docs(#issueId): Add README.md" can be used.
- [ ] Related file selection: Only relevant files should be touched and no other files should be affected.
- [ ] Format and Lint Suitability : The code should be made in accordance with a certain format standard and examined according to the lint rules.
- [ ] Clean commit history: The commits where the changes are made should be clear and organized.
- [ ] Opening PR when job is completed: PR should be opened when job is completed and sent for review by other team members.

## Changes

- [ ] A new feature (a change that adds a new feature, not a breaking change)
- [ ] A new refactor (a change that is not a breaking change, that improves the readability or performance of the code)
- [ ] A breaking change
- [ ] Documentation change

# How were these changes tested?

Please describe the tests you did to test the changes you made. Please also specify your test configuration.

**Test Configuration**:

* Firmware version:
* Hardware:
* Toolchain:
* SDK:

### CODE REVIEW

File: components/UI/FilterMenu/ChannelFilterMenu.tsx [hunk 1]
```diff
@@ -0,0 +1,24 @@
+import { useURLActions } from "@/stores/urlStore";
+import CommonFilterMenu, { FilterMenuOption } from "./CommonFilterMenu";
+
+type ChannelValueType = string | null;
+
+const channelFilterMenuOptions: FilterMenuOption<ChannelValueType>[] = [
+  { label: "all", value: null },
```
Comment (el) [inline components/UI/FilterMenu/ChannelFilterMenu.tsx:7]: I added an option for all channels here, but not sure if we should or not.

**Review by OrhanOzkercin — DISMISSED** [review]

**Review by fdemir — DISMISSED** [review]
lgtm

File: components/UI/Drawer/components/LayerContent.tsx [hunk 2]
```diff
@@ -65,6 +65,7 @@ export const LayerContent = () => {
             display: "flex",
             alignItems: "flex-start",
             margin: "0.5rem 0 0",
+            flexWrap: "wrap",
```
Comment (el) [inline components/UI/Drawer/components/LayerContent.tsx:68]: Fixes wrap issue

**Review by OrhanOzkercin — APPROVED** [review]
Eline saglik 🙏

**Pull request merged**